task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -436,6 +448,36 @@ def run(ti: RuntimeTaskInstance, log: Logger):
         SUPERVISOR_COMMS.send_request(msg=msg, log=log)
 
 
+def _push_xcom_if_needed(result: Any, ti: RuntimeTaskInstance):
```

Comment:
Any reason this isn't a method on RuntimeTI?

Reply:
I thought about that but avoided it as it will then be accessible via the Context.. `{{ ti._push_xcom }}`. 

If you don't think that a big deal, happy to move it since it technically belongs there

Reply:
Ohhh hmmm right. Yes avoiding exposing it via the context makes sense, but I worry about how this will scale.

Lets keep an eye on it